pg_replicate/src/clients/delta.rs

```diff
@@ -0,0 +1,235 @@
+use chrono::{NaiveDate, Utc};
+use deltalake::arrow::datatypes::{DataType as ArrowDataType, Field, Schema, TimeUnit};
+use deltalake::operations::create::CreateBuilder;
+use deltalake::{kernel::DataType, DeltaOps, DeltaTableError};
+use std::{collections::HashMap, path::Path, sync::Arc};
+use tokio_postgres::types::Type;
+
+use crate::{
+    conversions::{table_row::TableRow, Cell},
+    table::{ColumnSchema, TableId, TableName, TableSchema},
+};
+use deltalake::arrow::array::{
+    Array, BooleanArray, Date32Array, Float32Array, Float64Array, Int32Array,
+    RecordBatch as DeltaRecordBatch, StringArray, TimestampMicrosecondArray, UInt32Array,
+};
+
+pub struct DeltaClient {
+    pub path: String,
+    pub table_schemas: Option<HashMap<TableId, TableSchema>>,
+    pub delta_schemas: Option<HashMap<String, Arc<Schema>>>,
```

Comment:
The `table_schemas` and `delta_schema` can simply be `HashMap`s instead of `Option<HashMaps>` right?

Reply:
Use option because in firts init those value will empty first. their value will filled after some process.